### PR TITLE
[COST-3809] external id when using assume role

### PR DESCRIPTION
**Pull request by adberglund**
## Jira Ticket

[COST-3809](https://issues.redhat.com/browse/COST-3809)

## Description

This change will ...

## Testing

1. Checkout Branch
2. Restart Koku
3. Hit endpoint or launch shell
    1. You should see ...
4. Do more things...

## Notes

...

### CODE REVIEW

File: koku/masu/util/aws/common.py [hunk 1]
```diff
@@ -135,21 +135,22 @@ class AwsArn:
...
     resource = None
 
-    def __init__(self, arn):
+    def __init__(self, credentials):
         """
         Parse ARN string into its component pieces.
 
         Args:
-            arn (str): Amazon Resource Name
+            credential_json (Dict): Amazon Resource Name + External ID in credentials
```
Comment (adberglund) [inline koku/masu/util/aws/common.py:143]:
```suggestion
            credentials (Dict): Amazon Resource Name + External ID in credentials
```

**Review by lcouzens — APPROVED** [review]
Sweeeet!

**Comment by lcouzens** [comment]
Sorry Andrew for the mergy mergy conflicts but you knew it was coming :smile_cat:

**Review by lcouzens — APPROVED** [review]

**Pull request merged**